src/Tizen.NUI.Components/Controls/ViewList.cs

```
@@ -0,0 +1,302 @@
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        class InternalViewList : IList<View>
+        {
+            private List<View> internalViewList;
+            private ViewList viewList;
+
+            /// <summary>
+            /// Creates a new instance of a InternalViewList.
+            /// </summary>
+            /// <param name="parent">The ViewList object which has this InternalViewList</param>
+            [EditorBrowsable(EditorBrowsableState.Never)]
+            public InternalViewList(ViewList parent)
+            {
+                internalViewList = new List<View>();
+                viewList = parent;
+            }
+            /// <summary>
+            /// Get the View object by the index.
+            /// </summary>
+            /// <param name="index"></param>
+            [EditorBrowsable(EditorBrowsableState.Never)]
+            public View this[int index]
+            {
+                get
+                {
+                    return internalViewList[index];
+                }
+                [EditorBrowsable(EditorBrowsableState.Never)]
+                set
+                {
+                }
+            }
+            /// <summary>
+            /// Get the index of a View.
+            /// </summary>
+            /// <param name="view">The View object to get the index of</param>
+            /// <returns>The index</returns>
+            [EditorBrowsable(EditorBrowsableState.Never)]
+            public int IndexOf(View view)
+            {
+                return internalViewList.IndexOf(view);
+            }
+            /// <summary>
+            /// Insert a View in the list.
+            /// </summary>
+            /// <param name="order"></param>
+            /// <param name="view"></param>
+            [EditorBrowsable(EditorBrowsableState.Never)]
+            public void Insert(int order, View view)
+            {
+                if (order > internalViewList.Count)
+                {
+                    return;
```

Please throw exception for unexpected behavior.